Framework/Algorithms/src/AbsorptionCorrection.cpp

```diff
@@ -35,11 +35,31 @@ using namespace Kernel;
 using namespace Mantid::PhysicalConstants;
 using namespace Mantid::DataObjects;
 
+namespace {
+// the maximum number of elements to combine at once in the pairwise summation
+constexpr size_t MAX_INTEGRATION_LENGTH{1000};
+
+inline size_t findMiddle(const size_t start, const size_t stop) {
+  size_t half =
+      static_cast<size_t>(floor(.5 * (static_cast<double>(stop - start))));
+  return start + half;
+}
+
+double energyToWavelength(const double energyFixed) {
+  Unit_const_sptr energy = UnitFactory::Instance().create("Energy");
+  double factor, power;
+  energy->quickConversion(*UnitFactory::Instance().create("Wavelength"), factor,
+                          power);
+  return factor * std::pow(energyFixed, power);
```

Comment:
There is a `E_mev_toNeutronWavenumberSq` in `PhysicalConstants.h`. Could this function be shortened to
```c++
return std::sqrt(energyFixed / E_mev_toNeutronWavenumberSq);
```
?

Reply:
Good point.